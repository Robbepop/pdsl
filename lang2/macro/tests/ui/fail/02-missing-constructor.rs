#![feature(proc_macro_hygiene)]

use ink_lang2 as ink;

#[ink::contract(
    env = DefaultSrmlTypes,
    version = "0.1.0",
)]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
