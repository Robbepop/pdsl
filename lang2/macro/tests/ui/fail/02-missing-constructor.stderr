error: ink! contracts require at least one `#[ink(constructor)]`
  --> $DIR/02-missing-constructor.rs:6:1
   |
6  | / mod noop {
7  | |     #[ink(storage)]
8  | |     struct Noop {}
9  | |
...  |
13 | |     }
14 | | }
   | |_^
